src/Microsoft.AspNet.Mvc.Core/ActionResults/FilePathResult.cs

```diff
@@ -122,7 +121,12 @@ internal string ResolveFilePath(IFileProvider fileProvider)
             {
                 // The path is relative and IFileProvider found the file, so return the full
                 // path.
-                return fileInfo.PhysicalPath;
+                return new ResolveFilePathResult()
+                {
+                    // Note that physical path could be null in case of non-disk file (ex: embedded resource)
+                    PhysicalFilePath = fileInfo.PhysicalPath,
+                    FileInfo = fileInfo
+                };
             }
```

Comment:
Yeah, having said what I said, we should probably not refactor it if there's a real reason here, just document it.